pulser-core/pulser/sequence.py

```diff
@@ -1290,31 +1292,47 @@ def _precheck_target_qubits_set(
                 "qubits at a time."
             )
 
-        if self.is_parametrized():
-            for q in qubits_set:
-                if q not in self._qids and not isinstance(q, Parametrized):
-                    raise ValueError(
-                        "All non-variable qubits must belong to the register."
-                    )
-
         return qubits_set
 
     def _target(
         self,
-        qubits: Union[Iterable[QubitId], QubitId, Parametrized],
+        qubits: Union[Iterable[QubitId], QubitId],
         channel: str,
     ) -> None:
         qubits_set = self._precheck_target_qubits_set(qubits, channel)
-        if not self.is_parametrized():
+
+        if self.is_parametrized():
+            for q in qubits_set:
+                if q not in self._qids and not isinstance(q, Parametrized):
+                    raise ValueError(
+                        f"All non-variable qubits must belong to the register."
+                        f" Wrong id: {q!r}"
+                    )
```

Comment:
Isn't this still allowing the qubits in `qubits_set` to be parametrized? I think the check needs to be done regardless of whether the sequence is parametrized or not

Reply:
Yes, but what is the `not isinstance(q, Parametrized)` doing?

Reply:
Got it, thanks for the remark!